src/Program/View/CityView.hs

```diff
@@ -122,15 +124,20 @@ instance Drawable City where
                       ( attrLoc prog "aVertexPosition"
                       , Just ( attrLoc prog "aVertexNormal"
                              , attrLoc prog "aTextureCoord"))
-              setColor Nothing i obj = case behavior obj of
-                    Static  -> uniform4f gl colLoc 0.5 0.5 0.55 1
-                    Dynamic -> if i+1 == ai
-                               then uniform4f gl colLoc 1 0.6 0.6 1
-                               else uniform4f gl colLoc 0.75 0.75 0.7 1
+              setColor Nothing i obj = uniform4f gl colLoc r g b a
+                where
+                  (r, g, b, a) = unpackV4 $ case (behavior obj, i+1 == ai) of
+                                              (Static, _)      -> staticColor
+                                              (Dynamic, True)  -> activeColor
+                                              (Dynamic, False) -> getCityObjectColor blockColor obj
               setColor (Just arr) i obj = case unpackV4 $ PS.index i arr of
                     (r, g, b, a)  -> if behavior obj == Dynamic && i+1 == ai
                                      then uniform4f gl colLoc (g*0.5) (g*0.2) (b*0.2) a
                                      else uniform4f gl colLoc r g b a
+              blockColor = fromMaybe (vector4 0.75 0.75 0.7 1) $ (defaultBlockColor city) >>= convertHexToRGBA
```

Comment:
it is not very good performance-wise to do this unnecessary fromMaybe function every single frame. As I proposed earlier unwrap default colors from Maybe by moving this construct to the place where you parse colors.